app/src/main/java/com/tohami/newsapi/ui/news/NewsViewModel.java

```diff
@@ -0,0 +1,95 @@
+package com.tohami.newsapi.ui.news;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MediatorLiveData;
+import androidx.lifecycle.MutableLiveData;
+
+import com.tohami.newsapi.BuildConfig;
+import com.tohami.newsapi.Constants;
+import com.tohami.newsapi.base.BaseViewModel;
+import com.tohami.newsapi.data.model.DataModel;
+import com.tohami.newsapi.data.model.NewsArticle;
+import com.tohami.newsapi.data.model.NewsResponse;
+import com.tohami.newsapi.data.repos.NewsRepository;
+import com.tohami.newsapi.factory.RepositoryFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+class NewsViewModel extends BaseViewModel {
+
+    private final MediatorLiveData<DataModel<List<NewsArticle>>> newsArticlesLiveData = new MediatorLiveData<>() ;
+    private final MutableLiveData<NewsArticle> selectedArticleLiveData = new MutableLiveData<>() ;
+    private final NewsRepository newsRepository  ;
+
+    public NewsViewModel(RepositoryFactory repositoryFactory) {
+        super(repositoryFactory);
+        this.newsRepository = repositoryFactory.create(NewsRepository.class) ;
+    }
+
+    @Override
+    protected void onCleared() {
+        super.onCleared();
+        newsRepository.cancelAllRequests() ;
+    }
+
+
+    MediatorLiveData<DataModel<List<NewsArticle>>> getNewsArticlesLiveData() {
+        return newsArticlesLiveData;
+    }
+
+    void refreshNewsList() {
+        final LiveData<DataModel<NewsResponse>> theNextWeb =
+                newsRepository.getNews("the-next-web", BuildConfig.API_KEY);
+
+        final LiveData<DataModel<NewsResponse>> associatedPress =
+                newsRepository.getNews("associated-press", BuildConfig.API_KEY);
+
+        newsArticlesLiveData.addSource(
+                theNextWeb ,
+                newsResponseDataModel ->
+                        newsArticlesLiveData.setValue( combineLatestData(theNextWeb , associatedPress))
+        );
+
+        newsArticlesLiveData.addSource(
+                associatedPress ,
+                newsResponseDataModel ->
+                        newsArticlesLiveData.setValue( combineLatestData(theNextWeb , associatedPress))
+        );
+
+        newsArticlesLiveData.setValue(new DataModel<>(Constants.Status.LOADING , null , null));
+    }
+
+
+    private DataModel<List<NewsArticle>> combineLatestData(
```

Comment:
Use RXjava to combine the APIs, its a lot more powerful

Reply:
done